godtools/Managers/GTDataManager.swift

```diff
@@ -31,8 +41,40 @@ class GTDataManager: NSObject {
             .responseData()
     }
     
+    func rollbackContext() {
+        context.rollback()
+    }
+    
     func saveToDisk() {
-        NSManagedObjectContext.mr_default().mr_saveToPersistentStore(completion: nil)
+        saveToDisk(nil)
+    }
+    
+    func saveToDisk(_ completion: ((Bool, Error?) -> Void)?) {
+        context.mr_saveToPersistentStore(completion: completion)
+    }
+    
+    func saveToDiskAndWait() {
+        context.mr_saveToPersistentStoreAndWait()
+    }
+    
+    func findEntity<T: NSManagedObject>(_ entityClass: T.Type, byAttribute attribute: String, withValue value: Any) -> T? {
+        return entityClass.mr_findFirst(byAttribute: attribute, withValue: value, in: context)
+    }
+    
+    func findEntities<T: NSManagedObject>(_ entityClass: T.Type, matching: NSPredicate) -> [T] {
+        return entityClass.mr_findAll(with: matching, in: context) as! [T]
+    }
+    
+    func findAllEntities<T: NSManagedObject>(_ entityClass: T.Type) -> [T] {
+        return entityClass.mr_findAll(in: context) as! [T]
+    }
+    
+    func createEntity<T: NSManagedObject>(_ entityClass: T.Type) -> T? {
+        return entityClass.mr_createEntity(in: context)
+    }
+    
+    func deleteEntities<T: NSManagedObject>(_ entityClass: T.Type, matching: NSPredicate) {
```

Comment:
This function does match:
````swift
func findEntities<T: NSManagedObject>(_ entityClass: T.Type, matching: NSPredicate) -> [T]
````
which returns the entities "matching" the predicate. So it is consistent in that sense?

To me, `deleteAllEntities` would not take a predicate and just delete all records in that table.